operations/iuf/stages/management_nodes_rollout.md

```diff
@@ -26,22 +34,105 @@ The following arguments are most often used with the `management-nodes-rollout`
 
 ```bash
 iuf -a admin-230127 run --limit-management-rollout Management_Worker -r management-nodes-rollout
 ```
+
+Expected behavior: All NCN worker nodes will be rebuilt. Each set of worker nodes that is being rebuilt will contain 20% of the total number of worker nodes. For example, if there are 10 total worker nodes, then 2 will be rebuilt at a time.
+
+---
+
+(`ncn-m001#`) Execute the `management-nodes-rollout` stage for activity `admin-230127` using the following parameters. Upgrading the NCN master nodes as shown, should only be done if CSM is being upgraded.
+
+- Assume 10 worker nodes (`ncn-w001` through `ncn-w010`)
+- `-limit-management-rollout [Managment_Worker Management_Master]`
+- `-cmrp 35`
+- `ncn-w004` is labeled with `iuf-prevent-rollout=true`
+
+First, label `ncn-w004` and `ncn-m002` with `iuf-prevent-rollout=true`. Then execute the following command.
+
+```bash
+iuf -a admin-230127 run --limit-management-rollout Management_Worker Management_Master  --cmrp 33 -r management-nodes-rollout
+```
+
+Expected behavior:
+
+1. `ncn-m002` will be upgraded
+1. `ncn-m003` will be upgraded
+1. Worker nodes `ncn-w001,ncn-w002,ncn-w003` will be upgraded
+1. Worker nodes `ncn-w005,ncn-w006,ncn-w007` will be upgraded
+1. Worker nodes `ncn-w008,ncn-w009,ncn-w010` will be upgraded
+
+## Manually upgrade or rebuild NCN worker node with specific image and CFS configuration
+
+The upgrade and rebuild procedure for NCN worker nodes is the same. This section applies to both NCN worker node upgrades and NCN worker node rebuilds.
+The words 'rebuild' and 'upgrade' are exchangeable in this section.
+
+1. Get the image and CFS configuration created during `prepare-images` and `update-cfs-config` stages.
+Follow the instructions in [prepare-images](prepare_images.md#artifacts-created) to get the artifacts for `management-node-images`. For the image with the `configuration_group_name` matching
+`Management_Worker`, get the values for `final_image_id` and `configuration`.
+
+1. Upgrade/rebuild worker nodes
+
+    Worker nodes are automatically rebuilt using Argo workflows. If rebuilding multiple worker nodes at once, see [this page](../../node_management/Rebuild_NCNs/Rebuild_NCNs.md#restrictions) for restrictions.
+
+    (`ncn-m001#`) Rebuild worker node.
+
+    ```bash
+    /usr/share/doc/csm/upgrade/scripts/upgrade/ncn-upgrade-worker-storage-nodes.sh ncn-w001 --image-id <final_image_id> --desired-cfs-conf <configuration>
+    ```
+
+## Action needed if a worker rebuild fails
+
+In general, worker node rebuilds should complete successfully before starting another rebuild.
+The node can get into a bad state if it has been partially rebuilt and then it is attempted to restart the rebuild on that same node.
+In order to prevent this from happening, it is not possible to start another worker node rebuild if there is an incomplete worker node rebuild workflow.
```

Comment:
```suggestion
As a result, it is not possible to start another worker node rebuild if there is an existing incomplete worker node rebuild workflow.
```

Reply:
If you like this, then the following sentence needs to be deleted as I sucked it into this sentence.